clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts

```diff
@@ -64,19 +56,20 @@ export const {
   useDeleteDataQualifierMutation,
 } = dataQualifierApi;
 
+export interface State {}
+const initialState: State = {};
+
 export const dataQualifierSlice = createSlice({
   name: "dataQualifier",
   initialState,
-  reducers: {
-    setDataQualifiers: (state, action: PayloadAction<any>) => ({
-      ...state,
-      dataQualifier: action.payload,
-    }),
-  },
+  reducers: {},
 });
 
-export const { setDataQualifiers } = dataQualifierSlice.actions;
-export const selectDataQualifiers = (state: AppState) =>
-  state.dataQualifier.dataQualifier;
-
 export const { reducer } = dataQualifierSlice;
+
+const emptyDataQualifiers: DataQualifier[] = [];
+export const selectDataQualifiers: (state: AppState) => DataQualifier[] =
+  createSelector(
+    dataQualifierApi.endpoints.getAllDataQualifiers.select(),
```

Comment:
This returns a function. I explicitly wrote the return type `(state: AppState) => DataQualifier[]` for these selectors, but the type inference of `createSelector` and `api.endpoints.foo.select` actually does the trick on its own.